### ISSUE TITLE
How is the noSpawn hook supposed to work?

### ISSUE THREAD

**Issue by crowder** [avdg/screeps#3]
An example in your documentation would be awesome.

**Comment by avdg** [comment]
It's basically a hook that gets triggered when there is nothing to spawn when a spawn can spawn something.

**Comment by crowder** [comment]
Right, but what is it meant to return?

**Comment by avdg** [comment]
No, its just a hook with some parameters to execute the needed information. First being the spawn, the second being a function to spawn a creep even though you can use the spawn itself to make a creep spawn.

The custom spawn function is just to convert role information into a creep, which may need some documentation later on. But for the moment being, there are lots of other stuff I like to tweak as well and things can be very unstable (it's very fresh code).

It's being triggered here: https://github.com/avdg/screeps/blob/761999a0dc3d5d7bce9ffc6aaad8919def3805f0/extensions/tools/scripts/spawners.js#L123
